README.md

```diff
@@ -10,82 +10,110 @@ While xstate-tree manages your application state, it does not have a mechanism f
 
 At Koordinates we use xstate-tree for all new UI development. Our desktop application, built on top of [Kart](https://kartproject.org/) our Geospatial version control system, is built entirely with xstate-tree using GraphQL for global state.
 
-A minimal example of a single machine tree
+A minimal example of a single machine tree ([CodeSandbox](https://codesandbox.io/s/xstate-tree-b0el6e-b0el6e?file=/src/index.tsx)):
 
 ```tsx
 import React from "react";
 import { createRoot } from "react-dom/client";
 import { createMachine } from "xstate";
 import { assign } from "@xstate/immer";
-import { buildSelectors, buildActions, buildView, buildXstateTreeMachine, buildRootComponent } from "@koordinates/xstate-tree";
-
-type Events = { type: "SWITCH_CLICKED" } | { type: "INCREMENT", amount: number };
+import {
+  buildSelectors,
+  buildActions,
+  buildView,
+  buildXStateTreeMachine,
+  buildRootComponent
+} from "@koordinates/xstate-tree";
+
+type Events =
+  | { type: "SWITCH_CLICKED" }
+  | { type: "INCREMENT"; amount: number };
 type Context = { incremented: number };
 
 // If this tree had more than a single machine the slots to render child machines into would be defined here
 const slots = [];
 
 // A standard xstate machine, nothing extra is needed for xstate-tree
-const machine = createMachine<Context, Events>({
-  id: "root",
-  initial: "inactive",
-  context: {
-    incremented: 0,
-  },
-  states: {
-    inactive: {
-      on: {
-        SWITCH_CLICKED: "active",
-      },
+const machine = createMachine<Context, Events>(
+  {
+    id: "root",
+    initial: "inactive",
+    context: {
+      incremented: 0
     },
-    active: {
-      on: {
-        SWITCH_CLICKED: "idle",
-        INCREMENT: { actions: "increment" },
+    states: {
+      idle: {},
+      inactive: {
+        on: {
+          SWITCH_CLICKED: "active"
+        }
       },
-    },
+      active: {
+        on: {
+          SWITCH_CLICKED: "idle",
+          INCREMENT: { actions: "increment" }
+        }
+      }
+    }
   },
-}, {
-  actions: {
-    increment: assign((context, event) => {
-      context.incremented += event.amount;
-    }),
-  },
-});
+  {
+    actions: {
+      increment: assign((context, event) => {
+        context.incremented += event.amount;
```

Comment:
This doesn't type correctly without typegen, so it either needs to assert the event type is correct, guard against the event, or use typegen to have the event be typed as only `INCREMENT`